File: 8_structs/structs.c
#include <stdio.h>

/* Definir el tipo de dato stack usando struct */
typedef struct stack{
    int index;
    int data[5];
} stack;


/* Definir los metodos push y pull */
void push(stack *s, int k) {
    s->data[s->index] = k;
    s->index++;
}

int pull(stack *s) {
    s->index--;
    return s->data[s->index];
}

int main() {
    stack s; 
    s.index = 0;
    for (int i = 0; i < 5; i++){
        s.data[i] = 0;
    }
    stack *sptr = &s ;


    push(sptr, 23);
    push(sptr, 14);
    printf("stack: { %d ", s.index);
    printf("{ ");
    for (int i = 0; i < 5; i++){
        printf("%d ", s.data[i]);
    }
    printf("}}\n");
    return 0;
}

File: 08_structs/structs.go
package main

import "fmt"

/* Definir el tipo de dato stack usando struct */
type stack struct {
    index int
    data [5]int
}

/* Definir los metodos push y pop */
func (s *stack) push(k int) {
    s.data[s.index] = k
    s.index++
}

func (s *stack) pop() int {
    s.index--
    return s.data[s.index]
}

func main() {
    /* Crea un puntero a una nueva stack y haz push a dos valores */
    s := new(stack)
    s.push(23)
    s.push(14)
    fmt.Printf("stack: %v\n", *s) //stack: {2 [23 14 0 0 0]}
}


